pkg/dedup/iter.go

```diff
@@ -154,7 +160,11 @@ func (s *dedupSeriesSet) At() storage.Series {
 	repl := make([]storage.Series, len(s.replicas))
 	copy(repl, s.replicas)
 
-	return newDedupSeries(s.lset, repl, s.f)
+	if s.deduplicationFunc == AlgorithmChain {
+		return storage.ChainedSeriesMerge(s.replicas...)
```

Comment:
should we use ` storage.ChainedSeriesMerge(repl...)` instead? also fyi @yeya24 , do you recall why we need to make copy of `s.replicas` here but not when `len(s.replicas) == 1`?

Reply:
rebased and updated with a fix that replaces that line :)
sorry, I should have move it here faster

Reply:
no problem, do you know why a copy is necessary? `copy(repl, s.replicas)`

Reply:
no, I took the comment at the face value